### ISSUE TITLE
Handle Leak

### ISSUE THREAD

**Issue by narinishi** [Alex313031/thorium-legacy#64]
**System Details**
 - OS: Windows 8.1 (64-bit)
 - Thorium Version: 122.0.6261.132 (Official Build) (64-bit)

**Problem**
Process Hacker shows a very high handle count for one thorium.exe instance

**Screenshots**
![image](https://github.com/Alex313031/thorium-win7/assets/118132227/ad4da29b-24a8-4e4a-a4c0-3c47010aa907)

**Additional Notes**

**Comment by narinishi** [comment]
possibly related https://github.com/win32ss/supermium/issues/372

**Comment by narinishi** [comment]
The handle count was taken with no tabs open. However, I had previously been using the instance to browse various sites. I still have it open and the handle count has slightly increased.

I am also using Supermium 122.0.6261.85 (Official Build) (64-bit) and the handle count is nowhere near as high.

**Comment by Alex313031** [comment]
@narinishi Really? Because the handle leak in Thorium on my end on a Win 7 x64 machine is less than Supermium. Mine got to 11,000 after letting it sit for 3 days.

**Comment by narinishi** [comment]
> @narinishi Really? Because the handle leak in Thorium on my end on a Win 7 x64 machine is less than Supermium. Mine got to 11,000 after letting it sit for 3 days.

Yeah. I started a Thorium instance yesterday and it's already at 9,958 handles. Whereas Supermium instance running for two days is below 1,600. I did experience a memory leak with Supermium recently though, although its handles stayed normal.

**Comment by narinishi** [comment]
Came across a guide for determining handle leaks https://randomascii.wordpress.com/2021/07/25/finding-windows-handle-leaks-in-chromium-and-others/

**Comment by IDA-RE-things** [comment]
@narinishi , I think no one will fix this, because this dont gives likes from nubes, And not touching UI (so not visible ) ))
And requeser time to investigate and research.
More easy is to add UI feature and commandline flags., or another not comlex work.
So whats to wait ? :))

**Comment by narinishi** [comment]
There are two types of handles that Thorium is leaking. I have a process with 7002 total handles, 1641 are of type WaitCompletionPacket and 3487 are of type Event.

**Comment by JoachimHenze** [comment]
Progwrp 1.1.0.5010:
-A few function stubs added covering uxtheme and ole32, mostly for Windows 2000 targets
-Condition variable implementation rewritten, `handle leak eliminated`
[progwrp.zip](https://github.com/Alex313031/thorium-legacy/files/15282757/progwrp.zip)
Does that help you?

**Comment by narinishi** [comment]
> Not sure whether that would affect Win8.1x64 too, or whether the dll is entirely unused then.
Regular Thorium Legacy on 8.1 doesn't depend on progwrp, but I can try seeing if the XP version will run.

**Comment by narinishi** [comment]
![image](https://github.com/Alex313031/thorium-legacy/assets/118132227/54d48c93-b39a-4620-a3e4-e84fa5c44156)
Thorium for XP with latest progwrp isn't any better behaved - only a couple of hours and already near 15k handles

**Comment by Alex313031** [comment]
@JoachimHenze @narinishi He fixed this in latest progwrp. Wait until new builds come out tommorow, which will also have more bug fixes as well as FTP support!

**Comment by narinishi** [comment]
> @JoachimHenze @narinishi He fixed this in latest progwrp. Wait until new builds come out tommorow, which will also have more bug fixes as well as FTP support!
> 
> The new builds will use the latest 1.1.0.5010 dll

I was using the 1.1.0.5010 dll
![image](https://github.com/Alex313031/thorium-legacy/assets/118132227/ae53352d-8148-43a2-80dd-1417ef665a7a)

**Comment by Alex313031** [comment]
@narinishi @JoachimHenze Fixed in latest release > https://github.com/Alex313031/thorium-legacy/releases/tag/122.0.6261.171

**Comment by narinishi** [comment]
Still seems to occur 

![image](https://github.com/Alex313031/thorium-legacy/assets/118132227/6acd271a-2ad5-4829-a849-7f6652b4c46d)
![image](https://github.com/Alex313031/thorium-legacy/assets/118132227/da52c84e-1a6c-4a5c-a753-e731db6ec58c)

**Comment by IDA-RE-things** [comment]
you can check that this leak is not related to progwrp.dll at all, by using special version of progwrp for Win7+, which just redirects calls to system dll's

https://github.com/IDA-RE-things/Chrome-xp-api-adapter/releases

@narinishi : the issue should be reopened to be visible.
